metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/interfaces/MessageParser.java

```diff
@@ -34,15 +34,15 @@
    * @param rawMessage
    * @return If null is returned, this is treated as an empty list.
    */
-  List<T> parse(byte[] rawMessage);
+  List<T> parse(byte[] rawMessage, SensorParserConfig sensorParserConfig);
```

Comment:
I agree with @mattf-horton 's concerns here.  It is not ideal to change the interface.  

The GrokParser is a Configurable and so it has a configure method.  Can we use the configure method to pass through the necessary grok configuration?

The ParserBolt already calls parser.configure(...) in ParserBolt.prepare.  Can we have the ParserBolt.execute call parser.configure(...) anytime the configuration changes?  Or even each time, if you like?

It seems the Configurable interface was created to pass through configuration values, so it might be a bit cleaner this way.